### PR TITLE
BACK 1332 - Fetch pending transactions

### PR DESCRIPTION
**Pull request by mbasire-ledger**
## What is this about?

      * syncMempool step added to the synchronizeAccount job
      * UnconfirmedTransaction recorded by the Bookkeeper
      * InterpreterClient.saveUnconfirmedTransaction is stubbed
      * Eventual duplicated addresses removed before calling interpreter.compute

## Cute picture of lama
lama

### CODE REVIEW

File: coins/bitcoin/worker/src/test/scala/co/ledger/lama/bitcoin/worker/TransactionsCrawlerTest.scala [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+      .compile
+      .toList
+      .unsafeRunSync()
+
+    batchResults should have size 2
+    batchResults(0).transactions should contain only (transactions.values.toList.flatten: _*)
+    batchResults(1).addresses.map(_.accountAddress) should be(List())
+  }
+
+  it should "mark all met addresses as marked in the keychain" in {
```
Comment (nukky94) [inline coins/bitcoin/worker/src/test/scala/co/ledger/lama/bitcoin/worker/TransactionsCrawlerTest.scala:110]: typo

File: coins/bitcoin/worker/src/test/scala/co/ledger/lama/bitcoin/worker/TransactionsCrawlerTest.scala [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+      .saveUnconfirmedTransactions(Btc, accountId, keychainId, 20, 0, 20)
+      .stream
+      .compile
+      .toList
+      .unsafeRunSync()
+
+    keychain.newlyMarkedAddresses.toList should contain only ((11 to 17).map(_.toString): _*)
+  }
+
+  it should "send transactions and corresponding used addresses to the interpreter" in {}
```
Comment (nukky94) [inline coins/bitcoin/worker/src/test/scala/co/ledger/lama/bitcoin/worker/TransactionsCrawlerTest.scala:141]: nice test

File: coins/bitcoin/worker/src/test/scala/co/ledger/lama/bitcoin/worker/TransactionsCrawlerTest.scala [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+    0L,
+    1,
+    inputs.toList,
+    outputs.toList,
+    1
+  )
+
+}
+
+object Keychain {
```
Comment (nukky94) [inline coins/bitcoin/worker/src/test/scala/co/ledger/lama/bitcoin/worker/TransactionsCrawlerTest.scala:218]: `KeychainFixture`

**Review by nukky94 — CHANGES_REQUESTED** [review]

File: coins/bitcoin/worker/src/main/scala/co/ledger/lama/bitcoin/worker/models/BatchResult.scala [hunk 4]
```diff
@@ -1,10 +1,10 @@
 package co.ledger.lama.bitcoin.worker.models
 
-import co.ledger.lama.bitcoin.common.models.explorer.ConfirmedTransaction
+import co.ledger.lama.bitcoin.common.models.explorer.Transaction
 import co.ledger.lama.bitcoin.common.models.interpreter.AccountAddress
 
-case class BatchResult(
+case class BatchResult[TX <: Transaction](
```
[the remark this answers is not in the document]
Comment (mbasire-ledger) [reply coins/bitcoin/worker/src/main/scala/co/ledger/lama/bitcoin/worker/models/BatchResult.scala:6]: - [ ]

File: coins/bitcoin/worker/src/test/scala/co/ledger/lama/bitcoin/worker/TransactionsCrawlerTest.scala (hunk 1 above)
Comment (mbasire-ledger) [reply coins/bitcoin/worker/src/test/scala/co/ledger/lama/bitcoin/worker/TransactionsCrawlerTest.scala:110]: ok

File: coins/bitcoin/common/src/main/scala/co/ledger/lama/bitcoin/common/clients/http/ExplorerClient.scala [hunk 5]
```diff
@@ -55,6 +65,15 @@ class ExplorerHttpClient(httpClient: Client[IO], conf: ExplorerConfig, coin: Coi
       .expect[A](req)
       .handleErrorWith(e => IO.raiseError(ExplorerClientException(req.uri, e)))
 
+  private def callWithRetry[A](
+      req: Request[IO],
+      timeout: FiniteDuration
+  )(implicit cs: ContextShift[IO], c: EntityDecoder[IO, A], t: Timer[IO]): IO[A] =
+    IOUtils.retry(
```
[the remark this answers is not in the document]
Comment (mbasire-ledger) [reply coins/bitcoin/common/src/main/scala/co/ledger/lama/bitcoin/common/clients/http/ExplorerClient.scala:72]: A talk with explorer team let us suppose we can hit the explorer with no specific strategy. 
The `http 500 bug` (which may be fixed soon) may happen randomly and can not be fixed we a better strategy than recalling as soon as a 500 is returned.  The explorer can support `easily` 20 successive call.

**Review by nukky94 — APPROVED** [review]

**Pull request merged**